app/Http/Controllers/Forum/TopicsController.php

```diff
@@ -233,11 +234,22 @@ public function show($id)
             abort($skipLayout ? 204 : 404);
         }
 
-        $postsPosition = $topic->postsPosition($posts);
+        $firstPostId = $topic->posts()
+            ->showDeleted($showDeleted)
+            ->orderBy('post_id', 'asc')
+            ->first()->post_id;
+
+        // position of the first post, incremented in the view
+        // to generate positions of further posts
+        $firstPostPosition = $topic->postPosition($firstPostId);
```

Comment:
this probably should be position of the first loaded position?

(also there seems to be something wrong in "position of first post of the topic in the topic". Something with "first")